src/utils/importFuncGenerator.ts

```diff
@@ -0,0 +1,147 @@
+      return createFuncImport();
+    case STARKWARE_CAIRO_COMMON_BUILTINS + 'BitwiseBuiltin':
+      return createStructImport();
+    case STARKWARE_CAIRO_COMMON_BUILTINS + 'HashBuiltin':
+      return createStructImport();
+    case STARKWARE_CAIRO_COMMON_CAIRO_KECCAK + 'finalize_keccak':
+      return createFuncImport('range_check_ptr', 'bitwise_ptr');
+    case STARKWARE_CAIRO_COMMON_DEFAULT_DICT + 'default_dict_new':
+      return createFuncImport();
+    case STARKWARE_CAIRO_COMMON_DEFAULT_DICT + 'default_dict_finalize':
+      return createFuncImport('range_check_ptr');
+    case STARKWARE_CAIRO_COMMON_DICT + 'dict_read':
+      return createFuncImport('dict_ptr');
+    case STARKWARE_CAIRO_COMMON_DICT + 'dict_write':
+      return createFuncImport('dict_ptr');
+    case STARKWARE_CAIRO_COMMON_DICT_ACCESS + 'DictAccess':
+      return createStructImport();
+    case STARKWARE_CAIRO_COMMON_MATH + 'split_felt':
+      return createFuncImport('range_check_ptr');
+    case STARKWARE_CAIRO_COMMON_MATH_CMP + 'is_le':
+      return createFuncImport('range_check_ptr');
+    case STARKWARE_CAIRO_COMMON_MATH_CMP + 'is_le_felt':
+      return createFuncImport('range_check_ptr');
+    case STARKWARE_CAIRO_COMMON_UINT256 + 'Uint256':
+      return createStructImport();
+    case STARKWARE_CAIRO_COMMON_UINT256 + 'uint256_add':
+      return createFuncImport('range_check_ptr');
+    case STARKWARE_CAIRO_COMMON_UINT256 + 'uint256_eq':
+      return createFuncImport('range_check_ptr');
+    case STARKWARE_CAIRO_COMMON_UINT256 + 'uint256_le':
+      return createFuncImport('range_check_ptr');
+    case STARKWARE_CAIRO_COMMON_UINT256 + 'uint256_lt':
+      return createFuncImport('range_check_ptr');
+    case STARKWARE_CAIRO_COMMON_UINT256 + 'uint256_mul':
+      return createFuncImport('range_check_ptr');
+    case STARKWARE_CAIRO_COMMON_UINT256 + 'uint256_sub':
+      return createFuncImport('range_check_ptr');
+    case STARKWARE_STARKNET_COMMON_SYSCALLS + 'deploy':
+      return createFuncImport('syscall_ptr');
+    case STARKWARE_STARKNET_COMMON_SYSCALLS + 'emit_event':
+      return createFuncImport('syscall_ptr');
+    case STARKWARE_STARKNET_COMMON_SYSCALLS + 'get_caller_address':
+      return createFuncImport('syscall_ptr');
+    case STARKWARE_STARKNET_COMMON_SYSCALLS + 'get_contract_address':
+      return createFuncImport('syscall_ptr');
+    default:
+      assert(false, `Import ${name} from ${path} is not defined.`);
+      throw new TranspileFailedError(`Import ${name} from ${path} is not defined.`);
+  }
+}
+
+function findExistingImport(name: string, node: SourceUnit) {
+  const found = node.vFunctions.filter(
+    (n) => n instanceof CairoImportFunctionDefinition && n.name === name,
+  );
+
+  // assert(found.length < 2, `More than 1 import functions where found with name: ${name}.`);
+
+  if (found[0] !== undefined) {
+    assert(found[0] instanceof CairoImportFunctionDefinition);
+    return found[0];
+  }
+  return undefined;
```

Comment:
Why do you use that condition instead of asking for the length? The filtering should return only `CairoImportFunctionDefinition` nodes, so instead of making the assertion you could add it  in the filter method. It could look like the following:
```suggestion
function findExistingImport(name: string, node: SourceUnit) {
  const found = node.vFunctions.filter(
    (n) : n is CairoImportFunctionDefinition => n instanceof CairoImportFunctionDefinition && n.name === name,
  );

  // assert(found.length < 2, `More than 1 import functions where found with name: ${name}.`);

  return found.length !== 0 ? found[0] : undefined;
}
```

Reply:
+ 1

Reply:
Done